src/hooks/useEstimateRegistration.ts

```diff
@@ -0,0 +1,115 @@
+import { BigNumber } from 'ethers'
+import { toUtf8Bytes } from 'ethers/lib/utils'
+import { useMemo } from 'react'
+import { useFeeData, useQuery } from 'wagmi'
+
+import { formatsByCoinType, formatsByName } from '@ensdomains/address-encoder'
+
+import { RegistrationData } from '@app/components/pages/profile/[name]/registration/types'
+import { emptyAddress } from '@app/utils/constants'
+
+import useEstimateTransactionCost from './useEstimateTransactionCost'
+import { useNameDetails } from './useNameDetails'
+
+type RecordItem = { key: string; value: string }
+type RegistrationProps = {
+  reverseRecord: boolean
+  hasResolverSet: boolean
+  textRecords: RecordItem[]
+  addressRecords: RecordItem[]
+}
+type GasCostData = [index: number, gas: number]
+
+const byteLengthToDataInx = (byteLength: number) =>
+  byteLength > 1 ? Math.ceil(byteLength / 32) + 1 : byteLength
+
+const useEstimateRegistration = (data: RegistrationProps | undefined) => {
+  const { data: feeData, isLoading: feeDataLoading } = useFeeData()
+  const { data: gasCosts, isLoading: gasCostsLoading } = useQuery(['gas-costs'], async () => {
+    const addr = (await import('@app/assets/gas-costs/addr.json'))
+      .default as unknown as GasCostData[]
+    const text = (await import('@app/assets/gas-costs/text.json'))
+      .default as unknown as GasCostData[]
+
+    return { addr, text }
+  })
+
+  const estimate = useMemo(() => {
+    if (!feeData || !gasCosts || !data) return BigNumber.from(0)
+
+    const { addr, text } = gasCosts
+    const { reverseRecord, hasResolverSet, textRecords, addressRecords } = data
+
+    let limit = 240664
```

Comment:
Is the base cost for a registration? If so, let's name the variable or have a comment to be explicit about that